### PR TITLE
Feat: Added Firebase and Quiz Funci

### PR DESCRIPTION
**Pull request by Avin008**
### Includes:

- Authentication Features
- Quiz Features
- Leaderboard Feature

### CODE REVIEW

File: src/redux-toolkit/features/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  authStatus: localStorage.getItem("token") ? true : false,
+  token: localStorage.getItem("token"),
```
Comment (ganesh-kanchi) [inline src/redux-toolkit/features/authSlice.js:5]: you could use just 
`const initialState = {
    token: localStorage.getItem("token") || null
}`
and when you want to use auth status you could just substitute it with the token variable

File: src/components/singleQuizCard/SingleQuizCard.jsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          <button className="question-btn" onClick={() => navigate("/")}>
+            Quit
+          </button>
+          <button
+            className="question-btn next-question-btn"
+            onClick={() => nextQuestFunc()}
+          >
+            {data.singleQuizData.mcq.length <= quizState.questionNo + 1 &&
+            quizState.isAnswer
+              ? "View Reult"
```
Comment (ganesh-kanchi) [inline src/components/singleQuizCard/SingleQuizCard.jsx]: there is a typo here

**Review by ganesh-kanchi — APPROVED** [review]
the code is well written and readable, the functionalities are working fine, the file structure is also well maintained. this branch can be merged to dev.

File: src/redux-toolkit/features/authSlice.js (hunk 1 above)
Comment (Avin008) [reply src/redux-toolkit/features/authSlice.js:5]: tried it. but using authStatus to toggle between login and logout button in the navbar component and null !== false, so toggling was not working.

File: src/components/singleQuizCard/SingleQuizCard.jsx (hunk 2 above)
Comment (Avin008) [reply src/components/singleQuizCard/SingleQuizCard.jsx]: will fix it.

File: src/redux-toolkit/features/authSlice.js (hunk 1 above)
Comment (ganesh-kanchi) [reply src/redux-toolkit/features/authSlice.js:5]: alright then that's ok.

**Review by TusharJade — APPROVED** [review]
Awesome UI, functionalities are also working well. Everything is oerfect good to go for merge 👍🏻

**Pull request merged**